base/expr.jl

```diff
@@ -118,10 +118,42 @@ macro eval(x)
     :($(esc(:eval))($(Expr(:quote,x))))
 end
 
+"""
+    @inline
+
+Give a hint to the compiler that this function is worth inlining.
+
+Small functions typically don't even need the `@inline` annotation,
+it does it on its own. By using `@inline` on bigger functions,
+an extra nudge can be given to the compiler, like in the example below.
+```julia
+@inline function bigfunction(x)
+    #=
+        Function Definition
+    =#
+end
+```
+"""
 macro inline(ex)
     esc(isa(ex, Expr) ? pushmeta!(ex, :inline) : ex)
 end
 
+"""
+    @noinline
+
+Prevents the compiler from inlining a function.
+
+Small functions are typically automatically inlined.
+By using `@noinline` on small functions, auto-inlineing can be
+prevented like in the example below.
```

Comment:
Nit: the use of `like` here is reads informal. I'd probably separate this out into a separate sentence:
`[...] auto-inlining can be prevented. This is shown in the following example:`